### ISSUE TITLE
Include Heroic Modifiers in API call for milestones

### ISSUE THREAD

**Issue by mistahj67** [Bungie-net/api#473]
I've looked but haven't seen a way to determine the heroic strike modifiers via API, perhaps adding it to the milestone for heroic strikes similar to the nf milestone?

**Comment by vthornheart-bng** [comment]
Yes, I'd like to add this to the API, we indeed are currently missing this!

**Comment by vthornheart-bng** [comment]
TFS 654337

**Comment by ttgmpsn** [comment]
Since these are rotating daily, just like with #432, I'd love a way that is not tied to an active Milestone so that I can [keep getting the modifiers each day](http://redd.it/8i7lrk), regardless of Milestone completion.

**Comment by michabbb** [comment]
to be clear: so _right now_ there is not chance to get the heroic strike playlist modifiers anywhere, correct ?

**Comment by littleben4242** [comment]
This would be helpful.  Many folks in my clan would use this information to play their heroic strikes for the day.

**Comment by ttgmpsn** [comment]
@michabbb There is a way - it's the same as getting the daily challenges. See #432, parse the output of `https://www.bungie.net/en/Explore/Detail/DestinyActivityDefinition/<activityid>`. It will list challenges, and, if applicable, modifiers. For the heroic strikes, check https://www.bungie.net/en/Explore/Detail/DestinyActivityDefinition/4052671056

Parsing the website obviously isn't the perfect (or even a _good_) way, but it works as a basic workaround until proper ways to get Challenges & Modifiers are supported.

**Comment by michabbb** [comment]
@ttgmpsn many thanks! not optimal of course, but better than nothing ;)

**Comment by michabbb** [comment]
@vthornheart-bng don´t know where´s the right place to report this, but using this website would be much easiert, if

there wouldn´t be so many timeouts if you open [that page](https://www.bungie.net/en/Explore/Detail/DestinyActivityDefinition/4052671056) in a different language (like de)

![26-05-2018-01-19-01](https://user-images.githubusercontent.com/3524595/40575553-5bcf674a-60e7-11e8-9289-f94f2f01ff3f.png)

and if the rewards would be displayed correctly ;)

![26-05-2018-01-19-48](https://user-images.githubusercontent.com/3524595/40575563-78596320-60e7-11e8-8c5b-6cf069569ae1.png)

**Comment by michabbb** [comment]
in case someone might think it´s helpful, with the help of [this package](https://github.com/paquettg/php-html-parser), you can do this:

```ini
BUNGIE_WEBSITE_URL_STRIKE_PLAYLIST='https://www.bungie.net/en/Explore/Detail/DestinyActivityDefinition/4052671056'
BUNGIE_WEBSITE='https://www.bungie.net'
```


```php
$dom = new Dom;
		$dom->loadFromUrl(getenv('BUNGIE_WEBSITE_URL_STRIKE_PLAYLIST'),[
			'removeScripts' => true,
			'removeStyles' => true,
			'preserveLineBreaks' => true
		]);
		$contents = $dom->find('.grid-col-12-mobile');
		/** @var Dom\HtmlNode $content */
		foreach ($contents as $content) {
			if ($content->find('h3')->innerhtml==='Modifiers') {
				$modifiers = $content->find('.two-line-item-modifier-information');
				/** @var Dom\HtmlNode $modifier */
				foreach ($modifiers as $modifier) {
					$icon = $modifier->find('.icon')->getAttribute('style');
					$icon = strstr($icon,'?',true); //remove params from url
					$icon = getenv('BUNGIE_WEBSITE').substr($icon, strpos($icon, "'") + 1); //remove css
					echo $icon."\n";
					echo $modifier->find('.title')->innerhtml."\n";
					echo $modifier->find('.subtitle')->innerhtml."\n";
				}
			}
		}
```

**Comment by michabbb** [comment]
@vthornheart-bng omg... after the patch.... the last source of getting modifiers is broken, too ;-((

![30-05-2018-10-59-59](https://user-images.githubusercontent.com/3524595/40747310-2e628282-645d-11e8-9837-562b885ded98.png)

well done bungie 🤔

**Comment by floatingatoll** [comment]
This a good bug to report, but please don’t be mean to Bungie. They work incredibly hard on all of this for us and there’s only two or three of them total for an API that they don’t have to offer us at all.

Where does this screenshot originate - viewing PGCRs on the web, or in an official companion app, or elsewhere?

**Comment by michabbb** [comment]
@floatingatoll I don't want to be mean and I guess the people who are responsible for the website are not the same as for the api. The api people here are great, no question! So my frustration is not directed to the nice and helpful api guys here, but I cannot hide the frustration anyway, sorry. As things are not available via api you build workarounds and after the next patch, exact this workaround is no longer working, I pretty sure this is understandable.

So about your question, the question where this Screenshot is from, read the last postings in this issue, it's all about the same page which was mentioned to be used as the workaround.

**Comment by michabbb** [comment]
and [there is a workaround](https://www.reddit.com/r/DestinyTheGame/comments/8n1tww/predictive_heroic_strike_modifiers_through_feb/
) for the workaround 🙈
